Sources/Instrumentation/InstrumentationSystem.swift

```diff
@@ -58,13 +62,20 @@ public enum InstrumentationSystem {
     public static var instrument: Instrument {
         self.lock.withReaderLock { self._instrument }
     }
+}
 
-    // FIXME: smarter impl
-    public static var tracer: TracingInstrument {
+extension InstrumentationSystem {
+    /// Get an `Instrument` instance of the given type.
+    ///
+    /// When using `MultiplexInstrument`, this returns the first instance of the given type stored in the `MultiplexInstrument`.
+    /// - Parameter instrumentType: The type of `Instrument` you want to retrieve an instance for.
+    /// - Returns: An `Instrument` instance of the given type or `nil` if no `Instrument` of that type has been bootstrapped.
+    public static func instrument<I>(of instrumentType: I.Type) -> I? {
```

Comment:
not just any random `I`, but at least an instrument:

```suggestion
    public static func instrument<I>(of instrumentType: I.Type) -> I? where I: Instrument {
```

right?

Reply:
I did this at first but had to drop it because of:
> Value of protocol type 'TracingInstrument' cannot conform to 'Instrument'; only struct/enum/class types can conform to protocols

Reply:
can you share more context? We can't write what then; `of: TracingInstrument.self`?

if so, we could do another overload for T: TracingInstrument?

Reply:
#### `instrument<I: Instrument>(of:I)`

```swift
extension InstrumentationSystem {
    public static func instrument<I>(of instrumentType: I.Type) -> I? where I: Instrument {
        self.lock.withReaderLock {
            if let multiplexInstrument = self._instrument as? MultiplexInstrument {
                return multiplexInstrument.firstInstance(of: I.self)
            }
            return self._instrument as? I
        }
    }
}
```

#### `InstrumentationSystem.tracingInstrument`

```swift
extension InstrumentationSystem {
    public static var tracingInstrument: TracingInstrument {
        instrument(of: TracingInstrument.self)!
    }
}
```

> Value of protocol type 'TracingInstrument' cannot conform to 'Instrument'; only struct/enum/class types can conform to protocols

> if so, we could do another overload for T: TracingInstrument?

Would this be possible in the `TracingInstrumentation` library without exposing `.lock`?